src/components/dashboard/components/progressCard.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import styles from './progressCard.module.css'
+import LineChart from '../../common/charts/lineChart'
+import Button from './button'
+import Dropdown from './dropdown'
+import Labels from './labels'
+
+export default class ProgressCard extends Component {
+  state = {
+    // Show Weekly progress by default
+    weekly: true,
+    monthly: false,
+    yearly: false,
+    labelData: [
+      {
+        title: 'WPM',
+        color: '#F0A500',
+      },
+      {
+        title: 'Accuracy',
+        color: '#EC486F',
+      },
+    ],
+    labelOrientation: 'horizontal',
+    dropdownOptions: ['All', 'Practise', 'Arena'],
+  }
+
+  weeklyData = () => {
+    if (this.state.weekly === true) return
+    // console.log("Processing and loading weekly data!")
+    this.setState({
+      weekly: true,
+      monthly: false,
+      yearly: false,
+    })
+  }
+
+  monthlyData = () => {
+    if (this.state.monthly === true) return
+    // console.log("Processing and loading monthly data!")
+    this.setState({
+      weekly: false,
+      monthly: true,
+      yearly: false,
+    })
+  }
+
+  yearlyData = () => {
+    if (this.state.yearly === true) return
+    // console.log("Processing and loading yearly data!")
+    this.setState({
+      weekly: false,
+      monthly: false,
+      yearly: true,
+    })
+  }
+
+  render() {
+    return (
+      <div className={styles.card}>
+        <div className={styles.header}>
+          <Dropdown data={this.state.dropdownOptions} size="medium" />
+          <Labels
+            data={this.state.labelData}
+            orientation={this.state.labelOrientation}
+          />
+        </div>
+        <div className={styles.chartContainer}>
+          <LineChart width={870} height={300} />
+        </div>
+        <div className={styles.controls}>
+          <Button
+            text="Weekly"
+            state={this.state.weekly}
+            toggleData={this.weeklyData}
```

Comment:
Feel the state logic can change a bit. It can be a single state `duration` which can be a value from ("weekly", "monthly", "daily").

For button's state boolean: `this.state.duration === 'weekly'`.

For the toggleData prop, we can use [currying](https://javascript.info/currying-partials).
```
toggleDuration = (duration) => {
  return () => {
    this.setState({ duration: duration })
  }
}
...
<Button 
   state={this.state.duration === 'weekly'} 
   toggleData={this.toggleDuration('weekly')} />
```

Reply:
Current implementation method is not a good way to implement it?

Reply:
Yeah not the most efficient in this scenario. 
Currying is very nice in this scenarios. [example](https://www.nypl.org/blog/2016/12/06/currying-functions#:~:text=Using%20the%20currying%20technique%20is,rely%20on%20the%20currying%20concept.)

Reply:
Like all these states is basically like a boolean but with three values. So just one state is needed to represent the three.